docs/integration_fetching_data.md

```diff
@@ -0,0 +1,146 @@
+
+- `Entity.schedule_update_ha_state(force_refresh=False)`/`Entity.async_schedule_update_ha_state(force_refresh=False)` will schedule an update of the entity. If `force_refresh` is set to `True`, Home Assistant will call your entities update method (`update()`/`async_update()`) prior to writing the state.
+- `Entity.async_write_ha_state()` is an async callback that will write the state to the state machine within yielding to the event loop. It will not call your entity update method.
+
+# Polling API endpoints
+
+We're going to explain a few different API types here and the best way to integrate them in Home Assistant. Note that some integrations will encounter a combination of the ones below.
+
+## Coordinated, single API poll for data for all entities
+
+This API will have a single method to fetch data for all the entities that you have in Home Assistant. In this case we will want to have a single periodical poll on this endpoint, and then let entities know as soon as new data is available for them.
+
+Home Assistant provides a DataUpdateCoordinator class to help you manage this as efficiently as possible.
+
+```python
+import logging
+
+from homeassistant.helpers import debounce, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+
+_LOGGER = logging.getLogger(__name__)
+
+async def create_coordinator(api):
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await api.fetch_data()
+        except ApiError:
+            raise UpdateFailed
+
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="sensor",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=30),
+        # Debouncer to limit request_refresh
+        request_refresh_debouncer=debounce.Debouncer(
+            hass,
+            _LOGGER,
+            # Timeout to wait between refreshes
+            cooldown=0.3,
+            # When a refresh is requested, should we refresh right await and then wait
+            # Or should we wait and refresh at the end of timeout.
+            immediate=True
+        ),
+    )
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    return coordinator
+
+
+class MyEntity(entity.Entity):
+
+    def __init__(self, coordinator):
+        self.coordinator = coordinator
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        return not self.coordinator.failed_last_update
```

Comment:
```suggestion
        return not self.coordinator.last_update_failed
```

or maybe even skip the negation and use `updated_successfully` or similar to simplify this?